src/Orleans.CodeGenerator/SyntaxGeneration/FSharpUtils.cs

```diff
@@ -24,7 +24,16 @@ public static bool IsUnionCase(LibraryTypes libraryTypes, INamedTypeSymbol symbo
                 return false;
             }
 
-            if (!baseType.GetAttributes(compilationAttributeType, out var compilationAttributes) || compilationAttributes.Length == 0)
+            INamedTypeSymbol sumTypeCandidate;
+            if (symbol.GetAttributes(compilationAttributeType, out var compilationAttributes) && compilationAttributes.Length > 0)
+            {
+                sumTypeCandidate = symbol;
+            }
+            else if (baseType.GetAttributes(compilationAttributeType, out compilationAttributes) && compilationAttributes.Length > 0)
+            {
+                sumTypeCandidate = baseType;
+            }
```

Comment:
Do we ever need to worry about deeper inheritance chains than this?

Reply:
Not that I know of. There is however an issue with DUs of 4 cases or more, but not due to inheritance. I will open separate issue for that.